tests/countries/test_yemen.py

```diff
@@ -0,0 +1,237 @@
+#           dr-prodigy <dr.prodigy.github@gmail.com> (c) 2017-2023
+#           ryanss <ryanssdev@icloud.com> (c) 2014-2017
+#  Website: https://github.com/vacanza/holidays
+#  License: MIT (see LICENSE file)
+
+from unittest import TestCase
+
+from holidays.constants import SCHOOL, WORKDAY
+from holidays.countries.yemen import Yemen, YE, YEM
+from tests.common import CommonCountryTests
+
+
+class TestYemen(CommonCountryTests, TestCase):
+    @classmethod
+    def setUpClass(cls):
+        years = range(1990, 2050)
+        super().setUpClass(Yemen, years=years)
+        cls.no_estimated_holidays = Yemen(years=years, islamic_show_estimated=False)
+        cls.school_holidays = Yemen(categories=SCHOOL, years=years)
+        cls.workday_holidays = Yemen(categories=WORKDAY, years=years)
+        cls.workday_no_estimated_holidays = Yemen(
+            categories=WORKDAY,
+            years=years,
+            islamic_show_estimated=False,
+        )
+
+    def test_country_aliases(self):
+        self.assertAliases(Yemen, YE, YEM)
+
+    def test_no_holidays(self):
+        self.assertNoHolidays(Yemen(years=1990))
+
+    def test_labor_day(self):
+        name = "عيد العمال"
+        self.assertHolidayName(name, (f"{year}-05-01" for year in range(1991, 2050)))
+
+    def test_teachers_day(self):
+        name = "عيد المعلم"
+        self.assertNoHolidayName(name, self.school_holidays, range(1991, 2013))
+        self.assertHolidayName(
+            name, self.school_holidays, (f"{year}-05-05" for year in range(2013, 2050))
+        )
+
+    def test_unity_day(self):
+        name = "اليوم الوطني للجمهورية اليمنية"
+        self.assertHolidayName(name, (f"{year}-05-22" for year in range(1991, 2050)))
+
+    def test_victory_day(self):
+        name = "ذكرى 7 يوليو"
+        self.assertHolidayName(name, (f"{year}-07-07" for year in range(1991, 2000)))
+        self.assertHolidayName(
+            name, self.workday_holidays, (f"{year}-07-07" for year in range(2000, 2050))
+        )
+
+    def test_revolution_day(self):
+        name = "ثورة 26 سبتمبر المجيدة"
+        self.assertHolidayName(name, (f"{year}-09-26" for year in range(1991, 2050)))
+
+    def test_liberation_day(self):
+        name = "ثورة 14 أكتوبر المجيدة"
+        self.assertHolidayName(name, (f"{year}-10-14" for year in range(1991, 2050)))
+
+    def test_independence_day(self):
+        name = "عيد الجلاء"
+        self.assertHolidayName(name, (f"{year}-11-30" for year in range(1991, 2050)))
+
+    def test_hijri_new_year(self):
+        name = "عيد رأس السنة الهجرية"
+        dts = (
+            "2020-08-20",
+            "2021-08-10",
+            "2022-07-30",
+            "2023-07-19",
+            "2024-07-07",
+        )
+        self.assertHolidayName(name, dts)
+        self.assertHolidayName(name, self.no_estimated_holidays, range(1991, 2050))
+
+    def test_mawlid(self):
+        name = "المولد النبوي"
```

Comment:
```suggestion
        name = "المولد النبوي"
        self.assertNoHolidayName(name)
```
And same for `test_isra_and_miraj`.